allennlp_models/generation/dataset_readers/seq2seq.py

```diff
@@ -78,13 +78,29 @@ def __init__(
         self._target_tokenizer = target_tokenizer or self._source_tokenizer
         self._source_token_indexers = source_token_indexers or {"tokens": SingleIdTokenIndexer()}
         self._target_token_indexers = target_token_indexers or self._source_token_indexers
+
         self._source_add_start_token = source_add_start_token
         self._source_add_end_token = source_add_end_token
         self._target_add_start_token = target_add_start_token
         self._target_add_end_token = target_add_end_token
-        self._start_token, self._end_token = self._source_tokenizer.tokenize(
-            start_symbol + " " + end_symbol
-        )
+        self._start_token: Optional[Token] = None
+        self._end_token: Optional[Token] = None
+        if (
+            source_add_start_token
+            or source_add_end_token
+            or target_add_start_token
+            or target_add_end_token
+        ):
+            try:
+                self._start_token, self._end_token = self._source_tokenizer.tokenize(
+                    start_symbol + " " + end_symbol
```

Comment:
@epwalsh this will not work for character tokenizer. (even if `len(start_symbol) == 1` and `len(end_symbol) == 1`, because there is a space which is also a token)

any ideas how to fix this?

Reply:
In that case you could have your tokenizer add any special symbols while disabling this behavior in the dataset reader with `source_add_start_token=False`.